1_wqpdata/src/wqp_inventory.R

```diff
@@ -85,3 +82,59 @@ inventory_wqp <- function(ind_file, wqp_state_codes, wqp_states, wqp_codes) {
 
   invisible()
 }
+
+# prepare a plan for getting an inventory of available WQP data
+plan_inventory <- function(constituents, folders) {
+
+  inventory <- scipiper::create_task_step(
+    step_name = 'inventory',
+    target_name = function(task_name, step_name, ...) {
+      scipiper::as_ind_file(file.path(folders$tmp, sprintf('inventory_%s.feather', task_name)))
+    },
+    command = function(task_name, ...) {
+      sprintf("inventory_wqp(target_name, wqp_state_codes, wqp_states, wqp_codes, I('%s'))", task_name)
```

Comment:
This might be old code by now, but I was just thinking about this distinct inventory thing. `wqp_codes` is the read-in version of a yml file that contains all the codes for P, nitrate, temperature, and any other constituent groups we add, right? That means that if you add a constituent group, `wqp_codes` will change for all constituents groups, and the inventory will get rerun for all constituent groups. That's no good.

So I think we need another task_step before this one that subsets wqp_codes to just the codes relevant to a single constituent group. Then that subset of codes can get passed in to inventory_wqp in place of the current wqp_codes. This is the same sort of approach I took in subsetting the partitions for the data pull, and for the same reason.

Reply:
Ahhhh good catch! I will make sure to include that before I call this totally complete.

Reply:
So I should be able to add that subsetting as a target right above this line right? I would still be passing `wqp_codes` into `plan_inventory` but this plan would now include a step before `inventory_wqp` gets used and that would be passed this subset version.

Reply:
Yeah, I think that's right. The inventory plan can and should change if you add a constituent, but the call to inventory_wqp should stay unchanged. I think what you're suggesting would achieve that.

Reply:
OK, I think we did that in vizstorm and I followed that example. I thought we added the ability to use `[[` but maybe I'm wrong?

I edited the `plan_inventory` function so it produces this:

```
targets:
  # --- nitrate --- #
  
  characteristic_names_nitrate:
    command: wqp_codes[[I('characteristicName')]][[I('nitrate')]]

  1_wqpdata/tmp/data/inventory_nitrate.feather.ind:
    command: inventory_wqp(target_name, wqp_state_codes, wqp_states, wqp_codes, characteristic_names_nitrate, I('nitrate'))
```

But running `scipiper::scmake('characteristic_names_nitrate', remake_file = 'tasks_1_wqp_inventory.yml')` produces the `unknown special function [[` error.

Reply:
OK, two things.
* remake can handle one call to `[[` in a command, but it rejects two different calls to `[[` or any other function.
* wqp_codes is still going to change when you add temperature, which means that `inventory_wqp(target_name, wqp_state_codes, wqp_states, wqp_codes, ` is still an altered call even though you now also have `characteristic_names_nitrate` as another argument.

One solution that kills both problems with one function is to make that first target (currently `characteristic_names_nitrate`) be this instead:
```yml
wqp_codes_nitrate:
  command: subset_wqp_codes(wqp_codes, I('nitrate'))
```
and then write this function:
```r
subset_wqp_codes <- function(wqp_codes, constituent_group) {
  subset_codes <- wqp_codes
  subset_codes$characteristicName <- subset_codes$characteristicName[constituent_group]
  return(subset_codes)
}
```
and then rather than adding a new argument to the recipe for `1_wqpdata/tmp/data/inventory_nitrate.feather.ind`, as you've proposed above, just swap in `wqp_codes_nitrate` where you currently have `wqp_codes` (and then don't pass in `characteristic_names_nitrate` at all

Sound about right?

Reply:
Just about right except `wqp_codes` is needed for getting the site_type and sample_media out. Maybe those should just go in a separate cfg file?

Reply:
Agreed that we need that, but the `subset_wqp_codes` function above preserves `site_type` and `sample_media` in `wqp_codes_nitrate`, so I think my suggestion has that covered.

Reply:
ahhhh yes, sorry. Didn't take a close enough look at that. We are good for now 👍